src/src/rx-serial/SerialDisplayport.cpp

```diff
@@ -0,0 +1,81 @@
+/************************************************************************************
+Credits:
+  This software is based on and uses software published by Richard Amiss 2023,
+  QLiteOSD, which is based on work by Paul Kurucz (pkuruz):opentelem_to_bst_bridge 
+  as well as software from d3ngit : djihdfpv_mavlink_to_msp_V2 and 
+  crashsalot : VOT_to_DJIFPV
+
+THIS SOFTWARE IS PROVIDED IN AN "AS IS" CONDITION. NO WARRANTIES, WHETHER EXPRESS, 
+IMPLIED OR STATUTORY, INCLUDING, BUT NOT LIMITED TO, IMPLIED WARRANTIES OF 
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE APPLY TO THIS SOFTWARE. THE 
+COMPANY SHALL NOT, IN ANY CIRCUMSTANCES, BE LIABLE FOR SPECIAL, INCIDENTAL OR 
+CONSEQUENTIAL DAMAGES, FOR ANY REASON WHATSOEVER.
+************************************************************************************/
+
+#include "SerialDisplayport.h"
+#include "crsf_protocol.h"
+
+void SerialDisplayport::send(uint8_t messageID, void * payload, uint8_t size, Stream * _stream)
+{
+  _stream->write('$');
+  _stream->write('M');
+  _stream->write('<');
+  _stream->write(size);
+  _stream->write(messageID);
+  uint8_t checksum = size ^ messageID;
+  uint8_t * payloadPtr = (uint8_t*)payload;
+  for (uint8_t i = 0; i < size; ++i) {
+    uint8_t b = *(payloadPtr++);
+    checksum ^= b;
+    _stream->write(b);
+  }
+  _stream->write(checksum);
```

Comment:
There will be a small benefit in replacing the single byte writes with outputting the entire buffer at once e.g. write(buf, len);